pallet/src/lib.rs

```diff
@@ -79,12 +77,20 @@ pub mod pallet {
             session_for_validators_change: u32,
         ) -> DispatchResult {
             ensure_root(origin)?;
+            let current_session = Session::<T>::current_index();
+            // Because of how pallet `Session` works, it is not possible to make the change happen at
+            // `session_for_validators_change` if it is < `current_session` + 2. In order to store
+            // accurate information in our storage and give the user accurate feedback (in the form
```

Comment:
Something that has been bugging me a bit – by default I would return an error in such a case, since when you cannot do exactly what the user asks it's often better to avoid guessing. Is there a good reason in this case to assume that the user would want this change to happen at the earliest opportunity?

Reply:
Followup comment – would the obvious solution of adding a `force_change` boolean argument be possible? Is complicating the API worth it?

Reply:
Exactly, and return an error when cancelling.